### PR TITLE
fix: refund excess tokens

### PR DESCRIPTION
**Pull request by fedealconada**
fixes https://github.com/sherlock-audit/2023-03-sense-judging/issues/29

### CODE REVIEW

File: pkg/core/src/Periphery.sol [hunk 1]
```diff
@@ -446,6 +482,10 @@ contract Periphery is Trust, IERC3156FlashBorrower {
         permit2.permitTransferFrom(permit.msg, sigs, msg.sender, permit.sig);
         amt = _fromTarget(adapter, divider.combine(adapter, maturity, uBal), quote);
         _transfer(quote.buyToken, receiver, amt);
+
+        // refund any remaining underlying to receiver
+        ERC20 underlying = ERC20(Adapter(adapter).underlying());
+        _transfer(underlying, receiver, underlying.balanceOf(address(this)));
```
Comment (jparklev) [inline pkg/core/src/Periphery.sol]: Why not use `quote.sellToken` rather than calling the `underlying` getter on `adapter` here?

**Review by jparklev — COMMENTED** [review]
Thoughts on adding these excess refunds to `_toTarget` and `_fromTarget` directly?

File: pkg/core/src/Periphery.sol (hunk 1 above)
Comment (fedealconada) [reply pkg/core/src/Periphery.sol]: yeah, good call!

Comment (fedealconada) [reply pkg/core/src/Periphery.sol]: changed!

Comment (jparklev) [reply pkg/core/src/Periphery.sol]: awesome!

**Comment by jparklev** [comment]
> Thoughts on adding these excess refunds to `_toTarget` and `_fromTarget` directly?

gentle bump on this @fedealconada ^

**Comment by fedealconada** [comment]
> > Thoughts on adding these excess refunds to `_toTarget` and `_fromTarget` directly?
> 
> gentle bump on this @fedealconada ^

sorry! didn't see this. definitely! would be more optimal :) just pushed!

File: pkg/core/src/Periphery.sol [hunk 2]
```diff
@@ -1010,7 +1023,7 @@ contract Periphery is Trust, IERC3156FlashBorrower {
         address receiver,
         uint256 amt
     ) internal {
-        address(token) == ETH ? payable(receiver).transfer(amt) : token.safeTransfer(receiver, amt);
+        if (amt > 0) address(token) == ETH ? payable(receiver).transfer(amt) : token.safeTransfer(receiver, amt);
```
Comment (IAm0x52) [inline pkg/core/src/Periphery.sol]: The should be call rather than transfer for ETH

**Review by IAm0x52 — APPROVED** [review]
Fixes look good. Excess sell tokens are now returned inside _toTarget and _fromTarget

**Review by IAm0x52 — APPROVED** [review]
Fix looks good. ETH transfer changed to call

**Pull request merged**